[bugfix] Correctly import XQuery library modules from JAR files

adamretter opened the pull request.

dizzzz commented on src/org/exist/source/SourceFactory.java:
```diff
@@ -58,19 +64,26 @@
+        }
         /* file:// or location without scheme is assumed to be a file */
-        if(location.startsWith("file:") || location.indexOf(':') == Constants.STRING_NOT_FOUND)
+        else if(location.startsWith("file:") || location.indexOf(':') == Constants.STRING_NOT_FOUND)
```
Constants.STRING_NOT_FOUND ... can be replaced with a Java6+ operation on the string object?

adamretter replied:
This was existing code which I didn't modify.

dizzzz commented on src/org/exist/source/SourceFactory.java:
```diff
@@ -58,19 +64,26 @@
-            if(f.canRead())
-            {
-
+            final File f = new File(contextPath, location);
```
no Java7+ Path ?

adamretter replied:
Likewise this was existing code that I didn't rewrite.

dizzzz commented:
my remarks are not critical.....

ljo commented:
Yes, looks good and comments valid.

adamretter commented:
@dizzzz @ljo I just appended a commit to this that rewrites a bunch more of the old code to `java.nio.file`, this at least addresses the concerns that you raised. Hopefully all is good now?
